Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XlgDebugger/XlgDebuggerState.cs

```diff
@@ -311,16 +334,21 @@ private ObjectInfo DirectoryArtifactInfo(DirectoryArtifact d)
 
             var name = d.Path.GetName(PathTable).ToString(StringTable);
             var kind = d.IsSharedOpaque ? "shared opaque" : d.IsOutputDirectory() ? "exclusive opaque" : "source";
+            var members = d.IsOutputDirectory()
+                ? Analyzer.GetDirMembers(d, PipGraph.GetProducer(d))
+                : d.PartialSealId > 0
+                    ? PipGraph.ListSealedDirectoryContents(d).Select(f => f.Path)
+                    : CollectionUtilities.EmptyArray<AbsolutePath>();
             return new ObjectInfo(
                 preview: $"{name} [{kind}]",
                 properties: new[]
                 {
                     new Property("Path", d.Path.ToString(PathTable)),
                     new Property("PartialSealId", d.PartialSealId),
                     new Property("Kind", kind),
-                    d.IsOutputDirectory() ? new Property("Producer", () => PipGraph.GetProducer(d)) : null,
-                    new Property("Consumers", PipGraph.GetConsumingPips(d.Path).ToArray()),
-                    d.PartialSealId > 0 ? new Property("Members", () => PipGraph.ListSealedDirectoryContents(d)) : null
+                    d.IsOutputDirectory() ? new Property("Producer", () => Analyzer.GetPip(PipGraph.GetProducer(d))) : null,
+                    new Property("Consumers", PipGraph.GetConsumingPips(d)),
+                    new Property("Members", members)

```

Comment:
I was debugging some apex builds when I realized that this wasn't working correctly